### PR TITLE
fix: make the network listener more stable

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Make the network listener more stable. 

**Description**

- When network listener stops it should have a room to start again 
- With current configuration, it trigger start/stop based on one value which makes it brittle and can raise some edge cases. 

**Steps to test or reproduce**

Run all tests.

### CODE REVIEW

**Comment by dapplion** [comment]
> can raise some edge cases.

Can you explain this edge cases in more detail? Also please elaborate a bit more on why this config is brittle

**Comment by nazarhussain** [comment]
If set the value to `5` the server will **stop** listening on `5` connections and **start** listening back on **4**. This feels like a boolean condition, rather than a range. 

The edge case I m referring to is if one connection is made and one connection is dropped simultaneously. So the moment listener stopped and listener started again. I opened a PR in `libp2p` to fix that edge case, which was causing the listener to keep listening even libp2p is stopped. This would not have triggered if we have reasonable range difference between start and close limit.

File: packages/beacon-node/src/network/libp2p/index.ts [hunk 1]
```diff
@@ -81,7 +81,12 @@ export async function createNodeJsLibp2p(
         backlog: 5,
         closeServerOnMaxConnections: {
           closeAbove: networkOpts.maxPeers ?? Infinity,
-          listenBelow: networkOpts.maxPeers ?? Infinity,
+          // To stop and start listening based on a fixed number of connections
+          // would make it brittle to the order in which connections are established.
+          // Instead, we stop listening when the number of connections is below the target peers
+          // given that target peers and max peers have a reasonable gap between them and
+          // target peers are always less than the max peers
+          listenBelow: networkOpts.targetPeers ?? -Infinity,
```
Comment (dapplion) [inline packages/beacon-node/src/network/libp2p/index.ts:89]: Does this value need to default to -Infinity or 0?

Comment (nazarhussain) [reply packages/beacon-node/src/network/libp2p/index.ts:89]: I believe both can work as connection count never gonna be below zero. But with the consistent approach of positive Infinity, I suggest to use the negative Infinity.

**Comment by wemeetagain** [comment]
> If set the value to 5 the server will stop listening on 5 connections and start listening back on 4. This feels like a boolean condition, rather than a range.

That was the intended behavior of setting both values high.
If we have maxPeers, we don't want to listen for more connections.
If we have less than maxPeers, we want to listen for more connections.

I don't think that is brittle, and if it is, we just need to fix the underlying implementation.

That said, its worth thinking about what kind of connection behavior we want, and viewing this PR on its own merits.

File: packages/beacon-node/src/network/libp2p/index.ts (hunk 1 above)
Comment (wemeetagain) [reply packages/beacon-node/src/network/libp2p/index.ts:89]: -Infinity is dangerous because it means that once closeAbove is hit, the listener will never turn back on and we won't ever get inbound connections.

**Comment by nazarhussain** [comment]
As the underlying issue is addressed by https://github.com/libp2p/js-libp2p/pull/2058, so closing this PR.
